src/lib/falcor/falcorUtils.js

```diff
@@ -40,95 +40,110 @@ export function validateFalcorPathSets(falcorPathSets) {
   return falcorPathSets;
 }
 
-function recurseOnCache(curObject, remainingKeySets) {
-  if (remainingKeySets.length === 0) {
-    return true;
-  }
-  const nextRemainingKeySets = remainingKeySets.slice(1);
-  let nextKeySet = remainingKeySets[0];
-  if (!(nextKeySet instanceof Array)) {
-    nextKeySet = [nextKeySet];
+function followPath(path, object) {
+  // If using dot notation obj.key.key.key
+  if (typeof path === "string") {
+    path = path.split('.');
   }
-  return nextKeySet.every((key) => {
-    if (key !== null && typeof key === "object") {
-      let start = 0;
-      if (key.hasOwnProperty("from")) {
-        start = key.from;
-      }
-      let end;
-      if (key.hasOwnProperty("to")) {
-        if (key.hasOwnProperty("length")) {
-          throw new Error("Falcor Range cannot have both 'to' and 'length' properties at falcor KeySet: " + JSON.stringify(key));
+  return path.reduce((currentObject, nextChild) => {
+    if (currentObject !== undefined && currentObject.hasOwnProperty(nextChild)) {
+      return currentObject[nextChild];
+    }
+
+    return undefined;
+  }, object);
+}
+
+export function pathSetsInCache(cache, falcorPathSets) {
+  function handleCheckingSingleKey(curObject, nextRemainingKeySets, key) {
+    if (!curObject.hasOwnProperty(key)) {
+      return false;
+    }
+    else {
+      const val = curObject[key];
+      if (val.$type) {
+        switch (val.$type) {
+          case "error":
+          case "atom":
+            return nextRemainingKeySets.length === 0;
+          case "ref":
+            return checkSinglePathSetInCache(followPath(val.value, cache), nextRemainingKeySets);
+          default:
+            throw new Error("pathSetsInCache encountered unexpected type. Type found was: " + val.$type);
         }
-        end = key.to;
-      }
-      else if (key.hasOwnProperty("length")) {
-        end = start+key.length-1;
       }
       else {
-        throw new Error("Falcor Range must have either 'to' or 'length' properties at falcor KeySet: " + JSON.stringify(key));
+        return checkSinglePathSetInCache(val, nextRemainingKeySets);
       }
-      for (let i = start; i <= end; i++) {
-        if (curObject.hasOwnProperty(i)) {
-          if (!recurseOnCache(curObject[i], nextRemainingKeySets)) {
-            return false;
+    }
+  }
+
+  function checkSinglePathSetInCache(curObject, remainingKeySets) {
```

Comment:
See this is confusing because it's called `checkSinglePathSetInCache`, but then the second argument is `remainingKeySets`. Maybe a comment clarifying what this does. Like does it check one or multiple things?